sigle/src/modules/editor/EditorHeader.tsx

```diff
@@ -26,10 +29,47 @@ export const EditorHeader = ({
   onOpenSettings,
   onSave,
 }: EditorHeaderProps) => {
+  const [scrollDirection, setScrollDirection] = useState<ScrollDirection>(null);
+  const [prevOffset, setPrevOffset] = useState(0);
   const { user } = useAuth();
 
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  });
+
+  const handleScroll = () => {
+    let scrollY = window.scrollY;
+    if (scrollY === 0) {
+      setScrollDirection(null);
+    }
+    if (scrollY > prevOffset) {
+      setScrollDirection('down');
+    } else if (scrollY < prevOffset) {
+      setScrollDirection('up');
+    }
+    setPrevOffset(scrollY);
+  };
+
   return (
-    <Flex as="header" justify="between" align="center">
+    <Flex
+      css={{
+        position: 'sticky',
+        transform: scrollDirection === 'down' ? 'translateY(-100%)' : 'none',
+        transition: 'transform 0.5s, padding 0.2s',
+        backgroundColor: '$gray1',
+        top: window.scrollY < 40 ? 'auto' : 0,
```

Comment:
@gregogun net sure whether this one is responsible for when the header appears on mobile, but when scrolling on mobile it took more scrolling to see the menu than on web.